app/controllers/oauth/authorizations_controller.rb

```diff
@@ -19,7 +19,13 @@ def store_current_location
   end
 
   def render_success
-    if skip_authorization? || (matching_token? && !truthy_param?('force_login'))
+    # FIXME: Find a better way to apply this validation: if the scopes only
+    # includes offline_access, then it's not valid, since offline_access doesn't
+    # actually give access to resources:
+    if pre_auth.scopes.all?('offline_access')
+      error = Doorkeeper::OAuth::InvalidRequestResponse.new(reason: :offline_access_only, missing_param: nil)
+      render :error, locals: { error_response: error }, status: 400
```

Comment:
This check is a little funky and the error isn't right yet, but we probably need to add a validation in that checks that there are other scopes requested besides the `offline_access` scope, since that scope alone does not give any access to resources.

Reply:
Had a bit of a think about this, and I'm not sure we really need it. Yeah, they'd issue an access token and refresh token without this, but those wouldn't allow access to anything.